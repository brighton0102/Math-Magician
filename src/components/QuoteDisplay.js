import './Quote.css';
import React, { useState, useEffect } from 'react';

const QuoteDisplay = () => {
  const [quote, setQuote] = useState('');
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState('');

  const apiKey = 'jr0tU0nGnD2M7zd11ivtyw==k5fmqysgvHiEBZB4';
  const category = 'future';

  useEffect(() => {
    fetch(`https://api.api-ninjas.com/v1/quotes?category=${category}`, {
      method: 'GET',
      headers: {
        'X-Api-Key': apiKey,
        'Content-Type': 'application/json',
      },
    })
      .then((response) => response.json())
      .then((data) => {
        if (data.length > 0) {
          setQuote(data[0].quote);
          setLoading(false);
        } else {
          setError('No quotes found for the specified category.');
          setLoading(false);
        }
      })
      .catch(() => {
        setError('An error occurred while fetching the quote.');
        setLoading(false);
      });
  }, [category]);

  return (
    <div className="display-quote">
      <h2>Quote of the Day</h2>
      {loading && <p>Loading...</p>}
      {error && <p>{error}</p>}
      {!loading && !error && <p>{quote}</p>}
    </div>
  );
};

export default QuoteDisplay;
