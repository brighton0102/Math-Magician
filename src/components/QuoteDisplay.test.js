import React from 'react';
import renderer from 'react-test-renderer';
import QuoteDisplay from './QuoteDisplay';

test('QuoteDisplay snapshot test', () => {
  const component = renderer.create(<QuoteDisplay />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
