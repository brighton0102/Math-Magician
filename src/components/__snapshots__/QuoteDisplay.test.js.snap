// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteDisplay snapshot test 1`] = `
<div
  className="display-quote"
>
  <h2>
    Quote of the Day
  </h2>
  <p>
    Loading...
  </p>
</div>
`;
